bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/A5_3F/A5_3F_7F_EltakoFSB.java

```diff
@@ -96,16 +96,24 @@ protected State convertToStateImpl(String channelId, String channelTypeId,
         State currentState = getCurrentStateFunc.apply(channelId);
 
         if (currentState != null) {
-            int direction = getDB_1() == MoveUp ? -1 : 1;
             int duration = ((getDB_3Value() << 8) + getDB_2Value()) / 10; // => Time in DB3 and DB2 is given
                                                                           // in ms
 
-            PercentType current = currentState.as(PercentType.class);
-            if (config != null && current != null) {
+            if (config != null) {
                 EnOceanChannelRollershutterConfig c = config.as(EnOceanChannelRollershutterConfig.class);
-                if (c.shutTime != -1 && c.shutTime != 0) {
-                    return new PercentType(Math.min(100, (Math.max(0, current.intValue()
-                            + direction * ((duration * PercentType.HUNDRED.intValue()) / c.shutTime)))));
+                if (duration == c.shutTime) {
+                    return getDB_1() == MoveUp ? PercentType.ZERO : PercentType.HUNDRED;
+                } else {
+                    PercentType current = PercentType.ZERO;
+                    if (currentState != UnDefType.UNDEF) {
```

Comment:
```suggestion
                    if (currentState instanceof PercentType) {
```

Reply:
Good point, thanks for the hint. Fixed it.